### PR TITLE
LibJS: A few more ThrowCompletionOr conversions

### PR DESCRIPTION
**Pull request by trflynn89**
I think these are the last usages of `JS_DECLARE/DEFINE_OLD_NATIVE_FUNCTION` in LibJS

### CODE REVIEW

File: Userland/Libraries/LibJS/Runtime/ObjectPrototype.cpp [hunk 1]
```diff
@@ -136,27 +136,27 @@ JS_DEFINE_OLD_NATIVE_FUNCTION(ObjectPrototype::to_string)
 }
 
 // 20.1.3.5 Object.prototype.toLocaleString ( [ reserved1 [ , reserved2 ] ] ), https://tc39.es/ecma262/#sec-object.prototype.tolocalestring
-JS_DEFINE_OLD_NATIVE_FUNCTION(ObjectPrototype::to_locale_string)
+JS_DEFINE_NATIVE_FUNCTION(ObjectPrototype::to_locale_string)
 {
     auto this_value = vm.this_value(global_object);
-    return TRY_OR_DISCARD(this_value.invoke(global_object, vm.names.toString));
+    return TRY(this_value.invoke(global_object, vm.names.toString));
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/ObjectPrototype.cpp:142]: No need for the `TRY` here.

File: Userland/Libraries/LibJS/Runtime/GlobalObject.cpp [hunk 2]
```diff
@@ -423,9 +423,9 @@ JS_DEFINE_OLD_NATIVE_FUNCTION(GlobalObject::parse_int)
 }
 
 // 19.2.1 eval ( x ), https://tc39.es/ecma262/#sec-eval-x
-JS_DEFINE_OLD_NATIVE_FUNCTION(GlobalObject::eval)
+JS_DEFINE_NATIVE_FUNCTION(GlobalObject::eval)
 {
-    return TRY_OR_DISCARD(perform_eval(vm.argument(0), global_object, CallerMode::NonStrict, EvalMode::Indirect));
+    return TRY(perform_eval(vm.argument(0), global_object, CallerMode::NonStrict, EvalMode::Indirect));
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/GlobalObject.cpp:428]: same here

File: Userland/Libraries/LibJS/Runtime/IteratorPrototype.cpp [hunk 3]
```diff
@@ -20,17 +20,17 @@ void IteratorPrototype::initialize(GlobalObject& global_object)
...
 {
 }
 
 // 27.1.2.1 %IteratorPrototype% [ @@iterator ] ( ), https://tc39.es/ecma262/#sec-%iteratorprototype%-@@iterator
-JS_DEFINE_OLD_NATIVE_FUNCTION(IteratorPrototype::symbol_iterator)
+JS_DEFINE_NATIVE_FUNCTION(IteratorPrototype::symbol_iterator)
 {
-    auto* this_object = TRY_OR_DISCARD(vm.this_value(global_object).to_object(global_object));
+    auto* this_object = TRY(vm.this_value(global_object).to_object(global_object));
     return this_object;
```
Comment (IdanHo) [inline Userland/Libraries/LibJS/Runtime/IteratorPrototype.cpp:34]: This can be just one line

**Review by IdanHo — COMMENTED** [review]
Nice, just a couple of nitpicks

**Comment by trflynn89** [comment]
> Nice, just a couple of nitpicks

All good suggestions :) fixed, thanks!

**Pull request merged**